mla/test_statistics.py

```diff
@@ -51,12 +61,31 @@ def _preprocess(self, event_model: _models.EventModel,
         splines = event_model.log_sob_gamma_splines(
             events[super_prepro_dict['drop_index']])
 
-        return {**super_prepro_dict, 'splines': splines}
+        return {**super_prepro_dict, 'splines': splines, 'gamma': self.gamma}
+
+
+def _get_sob_energy(params: np.ndarray, prepro: I3Preprocessing) -> np.array:
+    """Docstring"""
+    if 'gamma' in params.dtype.names:
+        gamma = params['gamma']
+    else:
+        gamma = prepro.gamma
+    return np.exp([spline(gamma) for spline in prepro.splines])
+
+
+def i3_sob(params: np.ndarray, prepro: I3Preprocessing) -> np.array:
+    """Docstring"""
+    sob = prepro.sob_spatial
```

Comment:
Here need to change to prepro.sob_spatial.copy()

Reply:
thanks! fixed